extension/httpfs/s3fs.cpp

```diff
@@ -586,6 +616,7 @@ unique_ptr<ResponseWrapper> S3FileSystem::PutRequest(FileHandle &handle, string
 
 unique_ptr<ResponseWrapper> S3FileSystem::HeadRequest(FileHandle &handle, string url, HeaderMap header_map) {
 	auto auth_params = static_cast<S3FileHandle &>(handle).auth_params;
+	url = url.substr(0, url.find('?'));
```

Comment:
maybe we can both deduplicate this by adding an argument such as `bool strip_query_params` to the `S3UrlParse` function?

Reply:
I've added this only in the HEAD and GET request, because for other methods (like POST and PUT) the query paramaters are necessary in S3UrlParse (namely, the query params that are added for the multipart upload etc). An alternative solution I can think of is to pass the method to the S3UrlParse and then only alter the url in case of HEAD or GET method. What do you think?